gtest_parallel.py

```diff
@@ -398,12 +453,42 @@ def record_test_time(self, binary, testname, runtime_ms):
   def write_to_file(self, save_file):
     "Write all the times to file."
     try:
-      with open(save_file, "wb") as f:
-        with gzip.GzipFile("", "wb", 9, f) as gzf:
-          cPickle.dump(self.__times, gzf, cPickle.HIGHEST_PROTOCOL)
+      with TestTimes.LockedFile(save_file, 'a+b') as fd:
+        times = TestTimes.__read_test_times_file(fd)
+
+        if times is None:
+          times = self.__times
+        else:
+          times.update(self.__times)
+
+        # We erase data from file while still holding a lock to it. This
+        # way reading old test times and appending new ones are atomic
+        # for external viewer.
+        fd.truncate()
+        with gzip.GzipFile(fileobj=fd, mode='wb') as gzf:
+          cPickle.dump(times, gzf, cPickle.HIGHEST_PROTOCOL)
     except IOError:
       pass  # ignore errors---saving the times isn't that important
 
+  @staticmethod
+  def __read_test_times_file(fd):
+    # We seek to the beginning of file because if it was opened with 'a' mode
+    # then the file's position is at the end of file.
+    fd.seek(0)
```

Comment:
If we always open LockedFile at position 0 this shouldn't be needed either. WDYT?